### PR TITLE
Pet 6.x

### PR DESCRIPTION
**Pull request by brecky**
Start PET System Not complet!! Working

### CODE REVIEW

File: cmake/platform/settings.cmake [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# set installation prefix
+if( PREFIX )
+  set(CMAKE_INSTALL_PREFIX "${PREFIX}")
```
Comment (tkrokli) [inline cmake/platform/settings.cmake]: missing LineFeed / NewLine after `endif` on the line below.

File: src/server/bnetserver/Authentication/AuthCodes.cpp [hunk 2]
```diff
@@ -24,7 +24,7 @@ namespace AuthHelper
     // List of client builds for verbose version info in realmlist packet
     static std::vector<RealmBuildInfo> const ClientBuilds =
     {
-        { 20490, 6, 2, 2, 'a' },
+        { 20574, 6, 2, 2, 'a' },
```
Comment (DDuarte) [inline src/server/bnetserver/Authentication/AuthCodes.cpp]: This change does not belong to this PR.

File: cmake/platform/settings.cmake [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# set installation prefix
```
Comment (DDuarte) [inline cmake/platform/settings.cmake]: This change does not belong to this PR.

File: src/server/game/Entities/Pet/Pet.cpp [hunk 4]
```diff
@@ -310,6 +321,22 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
 
     owner->SetMinion(this, true);
     map->AddToMap(this->ToCreature());
+	
+	// Pet Added
+	if (map->AddToMap(this->ToCreature()))
```
Comment (DDuarte) [inline src/server/game/Entities/Pet/Pet.cpp]: Duplicated AddToMap call

File: cmake/platform/settings.cmake (hunk 1 above)
Comment (brecky) [inline cmake/platform/settings.cmake]: Thanks

File: src/server/game/Entities/Pet/Pet.cpp [hunk 5]
```diff
@@ -310,6 +321,22 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
...
+	{
+		TC_LOG_DEBUG("entities.pet", "Pet Added Send");
+		WorldPackets::Pet::PetAdded addedpacket;
+
+		addedpacket.petSlot = uint32(fields[7].GetUInt8());
+		addedpacket.petNumber = petId;
+		addedpacket.petCreatureID = fields[1].GetInt32();
+		addedpacket.petDisplayID = fields[3].GetInt32();
+		addedpacket.petExperienceLevel = fields[5].GetInt32();
+		addedpacket.petNameLenght = uint8(fields[8].GetString().length());
```
Comment (DDuarte) [inline src/server/game/Entities/Pet/Pet.cpp]: You do not need to store the length since you already store the whole string...

File: src/server/game/Entities/Player/Player.cpp [hunk 6]
```diff
@@ -20236,33 +20237,47 @@ void Player::SendOnCancelExpectedVehicleRideAura()
...
 
-    TC_LOG_DEBUG("entities.pet", "Pet Spells Groups");
+	TC_LOG_DEBUG("entities.pet", "Pet Spells Groups");
 
-    CharmInfo* charmInfo = pet->GetCharmInfo();
+	CharmInfo* charmInfo = pet->GetCharmInfo();
 
-    WorldPacket data(SMSG_PET_SPELLS_MESSAGE, 8 + 2 + 4 + 4 + 4 * MAX_UNIT_ACTION_BAR_INDEX + 1 + 1);
-    data << pet->GetGUID();
+	WorldPacket data(SMSG_PET_SPELLS_MESSAGE, 16 + 2 + 2 + 4 + 1 + 1 + 2 + 4 * MAX_UNIT_ACTION_BAR_INDEX + 4 + 4 + 4);
```
Comment (DDuarte) [inline src/server/game/Entities/Player/Player.cpp]: Needs to be converted to packet classes. Any packet writing or reading outside of Read() and Write() packet class methods is deprecated.

File: src/server/game/Server/Packets/PetPackets.h [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+            WorldPacket const* Write() override;
+
+            int32 petSlot = 0;
+            int32 petNumber = 0; //lowGuid (42129178)
+            int32 petCreatureID = 42717; //default Wolf PEt Hunter
+            int32 petDisplayID = 903;
+            int32 petExperienceLevel = 1;
+            int8  petFlags = 0;
+			std::string petName = "Pet";
```
Comment (DDuarte) [inline src/server/game/Server/Packets/PetPackets.h]: What?

File: src/server/game/Server/Packets/PetPackets.h [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        class PetAdded final : public ServerPacket
+        {
+        public:
+            PetAdded() : ServerPacket(SMSG_PET_ADDED, 5 * 4 + 1 + 1 + 10) { }
+
+            WorldPacket const* Write() override;
+
+            int32 petSlot = 0;
+            int32 petNumber = 0; //lowGuid (42129178)
+            int32 petCreatureID = 42717; //default Wolf PEt Hunter
```
Comment (DDuarte) [inline src/server/game/Server/Packets/PetPackets.h]: The default should not be a specific creature.

**Comment by DDuarte** [comment]
This PR is in a terrible state. There are tabs everywhere and our coding standard is not followed at all.

http://collab.kpsn.org/pages/viewpage.action?pageId=10682946

File: src/server/game/Server/Packets/PetPackets.h [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+    {
+        //CMSG_PET_RENAME
+        class PetRename final : public ClientPacket
+        {
+        public:
+            PetRename(WorldPacket&& packet) : ClientPacket(CMSG_PET_RENAME, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid petGuid;
```
Comment (DDuarte) [inline src/server/game/Server/Packets/PetPackets.h]: The naming of these fields should follow https://gist.github.com/Shauren/310fe3f1fcbadc74e102 (in UpperCammelCase)

File: src/server/game/Server/Packets/PetPackets.h [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+#include "ObjectGuid.h"
+#include "Pet.h"
+
+class WorldObject;
+
+namespace WorldPackets
+{
+    namespace Pet
+    {
+        //CMSG_PET_RENAME
```
Comment (DDuarte) [inline src/server/game/Server/Packets/PetPackets.h]: Redundant comment, packet name already in the constructor

File: src/server/game/Handlers/PetHandler.cpp [hunk 11]
```diff
@@ -59,64 +60,64 @@ void WorldSession::HandleDismissCritter(WorldPacket& recvData)
...
+   // recvData >> guid1;                                     //pet guid
+   // recvData >> data;
+   // recvData >> guid2;                                     //tag guid
+   // Position
+   // recvData >> x;
+   // recvData >> y;
+   // recvData >> z;
+
+   // uint32 spellid = UNIT_ACTION_BUTTON_ACTION(data);
+   // uint8 flag = UNIT_ACTION_BUTTON_TYPE(data);             //delete = 0x07 CastSpell = C1
```
Comment (DDuarte) [inline src/server/game/Handlers/PetHandler.cpp]: If this code is no longer needed remove it entirely instead of commenting it out.

File: src/server/game/Entities/Player/Player.cpp (hunk 6 above)
Comment (brecky) [inline src/server/game/Entities/Player/Player.cpp]: This is Test after  add to packet classes

File: src/server/game/Entities/Pet/Pet.cpp (hunk 5 above)
Comment (brecky) [inline src/server/game/Entities/Pet/Pet.cpp]: Ok howto convert to uint8 ?

Comment (DDuarte) [inline src/server/game/Entities/Pet/Pet.cpp]: `data << uint8(Name.length());` in the `Write()` method ...?

File: cmake/platform/settings.cmake (hunk 1 above)
Comment (tkrokli) [inline cmake/platform/settings.cmake]: Anyway, like DDuarte said, this file (along with a couple of others) don't belong in this PR.
See if you can find a way to separate your PR files from the normal 6.x updates.

File: src/server/game/Server/Packets/PetPackets.h (hunk 7 above)
Comment (tkrokli) [inline src/server/game/Server/Packets/PetPackets.h]: @brecky : you need to find a way to select the default pet creature for the player character race.

**Comment by DDuarte** [comment]
Re-open this once the issues I pointed out above (INCLUDING TABS) are fixed.
